diff --git a/proofs/fstar/extraction/Sandwich.Tunnel.Cyclic_bundle_473584177.fst b/proofs/fstar/extraction/Sandwich.Tunnel.Cyclic_bundle_473584177.fst
index 6ba5197..5ebdc57 100644
--- proofs/fstar/extraction/Sandwich.Tunnel.Cyclic_bundle_473584177.fst
+++ proofs/fstar/extraction/Sandwich.Tunnel.Cyclic_bundle_473584177.fst
@@ -11,13 +11,6 @@ let _ =
   let open Sandwich.Implementation.Openssl3_impl.Tunnel.X509_verify_param in
   let open Sandwich.Tunnel.Tls in
   ()
-assume val configured: Sandwich_api_proto.Configuration.t_Configuration -> bool
-
-[@@ FStar.Tactics.Typeclasses.tcinstance]
-assume val missing_impl: Protobuf.Enums.t_Enum Sandwich_api_proto.Configuration.t_Implementation
-
-assume
-val set_verify_mode_called: verify_mode: Sandwich.Tunnel.Tls.t_VerifyMode -> Type0
 
 /// Convenient wrapper around a `SSL_CTX`.
 type t_SslContext = | SslContext : Core.Ptr.Non_null.t_NonNull Openssl3.t_ssl_ctx_st -> t_SslContext
@@ -591,7 +587,7 @@ let try_from
       with
       | Core.Result.Result_Ok ssl_ctx ->
         let ssl_ctx_wrapped:t_SslContext =
-          SslContext (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_ctx_st ssl_ctx)
+          t_SslContext (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_ctx_st ssl_ctx)
           <:
           t_SslContext
         in
@@ -934,7 +930,7 @@ let prepare_ssl (self: t_TunnelBuilder)
       with
       | Core.Result.Result_Ok ssl ->
         let ssl_wrapped:t_Ssl =
-          Ssl (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_st ssl) <: t_Ssl
+          t_Ssl (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_st ssl) <: t_Ssl
         in
         (match
             set_subject_alternative_names ssl_wrapped tunnel_verifier
