diff --git a/proofs/fstar/extraction/Sandwich.Tunnel.Rec_bundle_54252925.fst b/proofs/fstar/extraction/Sandwich.Tunnel.Rec_bundle_54252925.fst
index 82233dc..0e964c8 100644
--- proofs/fstar/extraction/Sandwich.Tunnel.Rec_bundle_54252925.fst
+++ proofs/fstar/extraction/Sandwich.Tunnel.Rec_bundle_54252925.fst
@@ -12,14 +12,6 @@ let _ =
   let open Sandwich.Tunnel.Tls in
   ()
 
-assume val configured: Sandwich_api_proto.Configuration.t_Configuration -> bool
-
-[@@ FStar.Tactics.Typeclasses.tcinstance]
-assume val missing_impl: Protobuf.Enums.t_Enum Sandwich_api_proto.Configuration.t_Implementation
-
-assume
-val set_verify_mode_called: verify_mode: Sandwich.Tunnel.Tls.t_VerifyMode -> Type0
-
 /// Convenient wrapper around a `SSL_CTX`.
 type t_SslContext = | SslContext : Core.Ptr.Non_null.t_NonNull Openssl3.t_ssl_ctx_st -> t_SslContext
 
@@ -169,6 +161,7 @@ val set_alpn_protocols':
     #v_S: Type0 ->
     #impl_995885649_: Type0 ->
     {| i2: Core.Convert.t_AsRef v_S string |} ->
+    {| i3: Core.Iter.Traits.Collect.t_IntoIterator impl_995885649_ |} ->
     self: t_SslContext ->
     protocols: impl_995885649_
   -> Core.Result.t_Result Prims.unit Sandwich.Error.t_Error
@@ -176,7 +169,10 @@ val set_alpn_protocols':
 let set_alpn_protocols
       (#v_S #impl_995885649_: Type0)
       (#[FStar.Tactics.Typeclasses.tcresolve ()] i2: Core.Convert.t_AsRef v_S string)
-     = set_alpn_protocols' #v_S #impl_995885649_ #i2
+      (#[FStar.Tactics.Typeclasses.tcresolve ()]
+          i3:
+          Core.Iter.Traits.Collect.t_IntoIterator impl_995885649_)
+     = set_alpn_protocols' #v_S #impl_995885649_ #i2 #i3
 
 /// Sets the default parameters for a SSL context.
 assume
@@ -266,16 +262,16 @@ let configuration_get_mode_and_options
     : Core.Result.t_Result (t_Mode & Sandwich_api_proto.Tls.t_TLSOptions) Sandwich.Error.t_Error =
   Core.Option.impl__ok_or #(t_Mode & Sandwich_api_proto.Tls.t_TLSOptions)
     #Sandwich.Error.t_Error
-    (Core.Option.impl__and_then #Sandwich_api_proto.Configuration.t_Opts
+    (Core.Option.impl__and_then #Sandwich_api_proto.Configuration.Configuration.t_Opts
         #(t_Mode & Sandwich_api_proto.Tls.t_TLSOptions)
-        (Core.Option.impl__as_ref #Sandwich_api_proto.Configuration.t_Opts
+        (Core.Option.impl__as_ref #Sandwich_api_proto.Configuration.Configuration.t_Opts
             configuration.Sandwich_api_proto.Configuration.f_opts
           <:
-          Core.Option.t_Option Sandwich_api_proto.Configuration.t_Opts)
+          Core.Option.t_Option Sandwich_api_proto.Configuration.Configuration.t_Opts)
         (fun opts ->
-            let opts:Sandwich_api_proto.Configuration.t_Opts = opts in
+            let opts:Sandwich_api_proto.Configuration.Configuration.t_Opts = opts in
             match opts with
-            | Sandwich_api_proto.Configuration.Opts_Client opt ->
+            | Sandwich_api_proto.Configuration.Configuration.Opts_Client opt ->
               Core.Option.impl__map #Sandwich_api_proto.Tls.t_TLSOptions
                 #(t_Mode & Sandwich_api_proto.Tls.t_TLSOptions)
                 (Core.Option.impl__and_then #Sandwich_api_proto.Tls.t_TLSClientOptions
@@ -315,7 +311,7 @@ let configuration_get_mode_and_options
                     (Mode_Client <: t_Mode), tls <: (t_Mode & Sandwich_api_proto.Tls.t_TLSOptions))
               <:
               Core.Option.t_Option (t_Mode & Sandwich_api_proto.Tls.t_TLSOptions)
-            | Sandwich_api_proto.Configuration.Opts_Server opt ->
+            | Sandwich_api_proto.Configuration.Configuration.Opts_Server opt ->
               Core.Option.impl__map #Sandwich_api_proto.Tls.t_TLSOptions
                 #(t_Mode & Sandwich_api_proto.Tls.t_TLSOptions)
                 (Core.Option.impl__and_then #Sandwich_api_proto.Tls.t_TLSServerOptions
@@ -380,7 +376,7 @@ val set_minimum_tls_version': self: t_SslContext -> version: Sandwich.Tunnel.Tls
       (requires
         exists c mode max_version tls_options.
           configured c /\
-          configuration_get_mode_and_options c ==
+          Sandwich.Tunnel.Tls.Support.configuration_get_mode_and_options c ==
           Core.Result.Result_Ok (mode, tls_options) /\
           (tls_options_get_min_max_tls_version tls_options == (version, max_version)))
       (fun _ -> Prims.l_True)
@@ -574,7 +570,7 @@ let try_from
     (match new_ssl_context ctx mode with
       | Core.Result.Result_Ok ssl_ctx ->
         let ssl_ctx_wrapped:t_SslContext =
-          SslContext (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_ctx_st ssl_ctx)
+          t_SslContext (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_ctx_st ssl_ctx)
           <:
           t_SslContext
         in
@@ -863,7 +859,7 @@ let prepare_ssl (self: t_TunnelBuilder)
     (match new_ssl self.f_ssl_ctx with
       | Core.Result.Result_Ok ssl ->
         let ssl_wrapped:t_Ssl =
-          Ssl (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_st ssl) <: t_Ssl
+          t_Ssl (Sandwich.Support.Pimpl.impl_2__as_nonnull #Openssl3.t_ssl_st ssl) <: t_Ssl
         in
         (match set_subject_alternative_names ssl_wrapped tunnel_verifier with
           | Core.Result.Result_Ok _ ->
diff --git a/proofs/fstar/extraction/Sandwich.Tunnel.Tls.fst b/proofs/fstar/extraction/Sandwich.Tunnel.Tls.fst
index 7cb7558..371b71f 100644
--- proofs/fstar/extraction/Sandwich.Tunnel.Tls.fst
+++ proofs/fstar/extraction/Sandwich.Tunnel.Tls.fst
@@ -42,12 +42,12 @@ type t_VerifyMode =
 let get_tls13_config (cfg: Sandwich_api_proto.Configuration.t_Configuration)
     : Core.Option.t_Option Sandwich_api_proto.Tls.t_TLSv13Config =
   match
-    Core.Option.impl__as_ref #Sandwich_api_proto.Configuration.t_Opts
+    Core.Option.impl__as_ref #Sandwich_api_proto.Configuration.Configuration.t_Opts
       cfg.Sandwich_api_proto.Configuration.f_opts
   with
   | Core.Option.Option_Some opts ->
     (match opts with
-      | Sandwich_api_proto.Configuration.Opts_Client c ->
+      | Sandwich_api_proto.Configuration.Configuration.Opts_Client c ->
         (match
             Core.Option.impl__as_ref #Sandwich_api_proto.Configuration.Client_options.t_Opts
               c.Sandwich_api_proto.Configuration.f_opts
@@ -74,7 +74,7 @@ let get_tls13_config (cfg: Sandwich_api_proto.Configuration.t_Configuration)
                     Rust_primitives.Hax.t_Never))
           | Core.Option.Option_None  ->
             Core.Option.Option_None <: Core.Option.t_Option Sandwich_api_proto.Tls.t_TLSv13Config)
-      | Sandwich_api_proto.Configuration.Opts_Server c ->
+      | Sandwich_api_proto.Configuration.Configuration.Opts_Server c ->
         (match
             Core.Option.impl__as_ref #Sandwich_api_proto.Configuration.Server_options.t_Opts
               c.Sandwich_api_proto.Configuration.f_opts
